diff --git a/MOVED b/MOVED
index c9242002b..87f6f3653 100644
--- a/MOVED
+++ b/MOVED
@@ -1662,7 +1662,6 @@ audio/aacplusenc||2022-09-30|Has expired: Legacy audio format, superseded by HE-
 net/libmms||2022-09-30|Has expired: Obsolete, Microsoft deprecated MMS protocol in favour of RTSP in 2003
 audio/mctoolame-decoder||2022-09-30|Has expired: Legacy and obscure audio format that never gained any traction
 audio/mctoolame-encoder||2022-09-30|Has expired: Legacy and obscure audio format that never gained any traction
-www/py-ws4py||2022-09-30|Has expired: Upstream abandoned
 databases/py-bsddb3|databases/py-berkeleydb|2022-09-30|Has expired: Requires upstream EOLd db5. Use py-berkeleydb
 graphics/megapov|graphics/povray37|2022-10-01|Has expired: Abandoned upstream. graphics/povray37 is the recommended replacement
 x11-toolkits/gstreamer1-plugins-gtk4||2022-10-01|No consumers left and deprecated in favor gtk4 plugin from multimedia/gstreamer1-plugins-rust >= 0.8.0
@@ -4871,7 +4870,6 @@ textproc/moar|textproc/moor|2025-12-01|Has expired: Upstream renamed to moor, us
 accessibility/gammy||2025-12-01|Has expired: Abandoned and unfetchable
 mail/mu4e-maildirs|mail/mu|2025-12-01|Has expired: Upstream discourages using this software and recommends feature now integrated into mail/mu
 devel/libavl||2025-12-04|Has expired: Since 2002 libavl is no longer distributed as a shared library
-net/py-netif||2025-12-04|Has expired: Upstream unmaintained
 x11-fonts/gentium-basic|x11-fonts/gentium|2025-12-04|Has expired: Superceeded by Gentium-7.000 https://software.sil.org/gentium/download/
 www/rubygem-cssbundling-rails-gitlab|www/rubygem-cssbundling-rails-rails-gitlab|2025-12-05|Obsolete, use www/rubygem-cssbundling-rails-rails-gitlab instead
 misc/amazon-qldb-shell||2025-12-05|Discontinued by Amazon as of July 2025: old and new customers are advised to migrate to Amazon Aurora PostgreSQL
