diff --git a/Mk/Uses/python.mk b/Mk/Uses/python.mk
index 5ef639cf7907..f8b9c38d08e0 100644
--- a/Mk/Uses/python.mk
+++ b/Mk/Uses/python.mk
@@ -605,11 +605,7 @@ _PYTHONPKGLIST=	${WRKDIR}/.PLIST.pymodtmp
 #
 
 # cryptography* support
-.  if ${PYCRYPTOGRAPHY_DEFAULT} == rust
-CRYPTOGRAPHY_DEPENDS=	${PYTHON_PKGNAMEPREFIX}cryptography>=41.0.7,1:security/py-cryptography@${PY_FLAVOR}
-.  else
 CRYPTOGRAPHY_DEPENDS=	${PYTHON_PKGNAMEPREFIX}cryptography-legacy>=3.4.8_1,1:security/py-cryptography-legacy@${PY_FLAVOR}
-.  endif
 
 .  if defined(_PYTHON_FEATURE_CRYPTOGRAPHY_BUILD)
 BUILD_DEPENDS+=	${CRYPTOGRAPHY_DEPENDS}
