diff --git a/Mk/Uses/waf.mk b/Mk/Uses/waf.mk
index 06b41c02304e..59bb87524857 100644
--- a/Mk/Uses/waf.mk
+++ b/Mk/Uses/waf.mk
@@ -30,14 +30,15 @@ ALL_TARGET=	#
 HAS_CONFIGURE=	yes
 MAKE_ARGS+=	--verbose
 WAF_CMD?=	./waf
+WAF_ENV?=	""
 
 CONFIGURE_TARGET?=	configure
 ALL_TARGET?=		build
 INSTALL_TARGET?=	install
 TEST_TARGET?=		test
 
-CONFIGURE_CMD=	${PYTHON_CMD} ${WAF_CMD} ${CONFIGURE_TARGET}
-MAKE_CMD=	${PYTHON_CMD} ${WAF_CMD}
+CONFIGURE_CMD=	${WAF_ENV} ${PYTHON_CMD} ${WAF_CMD} ${CONFIGURE_TARGET}
+MAKE_CMD=	${WAF_ENV} ${PYTHON_CMD} ${WAF_CMD}
 CONFIGURE_ARGS+=	--prefix=${PREFIX} \
 			${_MAKE_JOBS}
 
